### ISSUE TITLE
edge.js templates

### ISSUE THREAD

**Issue by johntom** [fastify/point-of-view#417]
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the feature has not already been requested

### 🚀 Feature Proposal

I just came across edge.js templates from "htmx discord link"  and think this would be a strong addition to point-of-view.

### Motivation

While I'm currently using nunjucks as my templating engine, many users on discord feel they will never add partials which is a feature I'd like to test with htmx. I also find https://edgejs.dev/docs/introduction very clear and concise.

**Comment by mcollina** [comment]
Thanks for reporting! Would you like to send a Pull Request to address this issue? Remember to add unit tests.

**Comment by johntom** [comment]
Yes I don't think I'll have any isses with a pull request(pr) but have not gotten edge to work with fastify so didnt have anything to contribute,
Do you want me to do the pr?

**Comment by johntom** [comment]
Ok, I think I have successfully created a pull request.

**Comment by mweberxyz** [comment]
I am not sure edge is a great fit for this plugin. Looking at https://edgejs.dev/docs/getting_started, it is ESM-only (whereas this entire plugin is CommonJS), and it looks like it has HTTP routing and middleware built in, just looking at this getting started example, I am not sure how Fastify fits in:

```
import { Edge } from 'edge.js'
import { createServer } from 'node:http'

const edge = Edge.create()
edge.mount(new URL('./views', import.meta.url))

const server = createServer(async (req, res) => {
  const data = { username: 'virk' }
  const html = await edge.render('home', data)

  res.setHeader('content-type', 'text/html')
  res.end(html)
})

server.listen(3000)
```

**Comment by johntom** [comment]
I never got edge to work and was off converting a Fastify3/ commonJS to
Fastify4/ESM. To that end, the following code will let you require files
with esm

'use strict'
import fp from 'fastify-plugin';
import os from 'os'
const drive = process.env.drive;

// The following fixes common libs when you cant import
import { createRequire } from "module";
const require = createRequire(import.meta.url);
// End of fix

console.log(" chilkat================== ")
 if (os.platform() == 'win32') {
  if (os.arch() == 'ia32') {
    var chilkat = ***@***.***/ck-node20-win-ia32');
  } else {
    var chilkat = ***@***.***/ck-node20-win64');
  }
}
// esm does not support loading json and if fixed with this snippet
  const books = require("../json/books.json")
